### ISSUE TITLE
You can find the codes here

### ISSUE THREAD

**Issue by chaman9000** [microsoft/NUWA#20]
As mentioned in #2, #4, #5, #6, #7, #9, #14, #15, #16, #19, they aren't going to give you a fuck, as if they care. Especially they are in China, a place disconnected from the world.

You can find the following codes to achieve the corresponding tasks (many of them also achieves superior quality).
And remember to cite them (instead of NUWA of course) to support of their efforts in making their codes publicly available.

 - Text to image: 
   - https://github.com/CompVis/stable-diffusion
 - Text to video: 
   - https://github.com/THUDM/CogVideo
   - https://github.com/lucidrains/imagen-pytorch
 - Infinite-visual synthesis: 
   - https://github.com/google-research/google-research/tree/master/infinite_nature_zero 
   - https://github.com/hubert0527/infinityGAN 
   - https://github.com/universome/alis

**Comment by aidotayyy** [comment]
How ridiculous you are。you're so stupid
Microsoft Research Asia is an  pure American company
Not open source is just a business freedom for the company, but when you rise to the country, you are a racist and shame on you,
curry muncher！Hahahah

**Comment by aidotayyy** [comment]
What a ridiculous Paki. I suggest you drink your Ganges River in India to improve your IQ so that you know that MSRA is an American company. Is the cow urine in your country good to drink?
I heard that you are a country rich in rapists, and they don't even let go of cows. , are they all shameless curry muncher like you

**Comment by lopho** [comment]
wtf

**Comment by Vancause** [comment]
The code (https://github.com/google-research/google-research/tree/master/infinite_nature_zero) can not be ran.

**Comment by wufeim** [comment]
> Especially they are in China, a place disconnected from the world.

@chaman9000 Nobody is obligated to share ANYTHING with you. Don't take code/checkpoints for granted and be a crying baby on the Internet.

There are tons of works by Google/Microsoft/Amazon that are not open-sourced, let alone the so-called OpenAI.

**Comment by chaman9000** [comment]
@wufeim Welp, then the author should not mention the codes will be released soon in https://github.com/microsoft/NUWA/issues/2#issuecomment-979948411. I guess your mom taught you keep your words, right?

Your logic: Some works show it is fine to not open-source their codes, so it is completely a good idea to join them and be the worst part of the community together!

**Comment by wufeim** [comment]
@chaman9000 Have some industrial experiences and then comment on open-sourcing models from big companies. Also look into recent cases and learn about legal concerns of releasing large generative models.

I hope you would join OpenAI one day and open-source everything!

**Comment by chaman9000** [comment]
@wufeim Right, it is clever to assume yourself is the best, and the only one who has the precious precious industrial experiences LMAO. But you are surely wrong this time, and having industrial experience is nothing particularly valuable.

The recent legal concern is just bullshit in the context of this years-old model not releasing the codes. Furthermore, what is the relation between "code" release and the legal concerns on the "trained model" LoL?

And don't try to shift the focus, the author should not pledge (https://github.com/microsoft/NUWA/issues/2#issuecomment-979948411) to share the codes if they are not planning to do so. There were papers having legal issues while releasing codes years back due to licenses or legal reviews, people keep their word by sharing their own re-implementations outside the company. It is just whether they are willing to be an honest human, or comfortable lying and being ignorant.

**Comment by jackiemoo** [comment]
> @wufeim Right, it is clever to assume yourself is the best, and the only one who has the precious precious industrial experiences LMAO. But you are surely wrong this time, and having industrial experience is nothing particularly valuable.
> 
> The recent legal concern is just bullshit in the context of this years-old model not releasing the codes. Furthermore, what is the relation between "code" release and the legal concerns on the "trained model" LoL?
> 
> And don't try to shift the focus, the author should not pledge ([#2 (comment)](https://github.com/microsoft/NUWA/issues/2#issuecomment-979948411)) to share the codes if they are not planning to do so. There were papers having legal issues while releasing codes years back due to licenses or legal reviews, people keep their word by sharing their own re-implementations outside the company. It is just whether they are willing to be an honest human, or comfortable lying and being ignorant.

'The model and code will be made public in the near future. But recently we need time to clean up the code to make it simple and easy to reproduce. In addition, we also need to follow the company's public process. '
He said that he would open source the code, but he also said that he must abide by the company's open source process, and his personal opinions cannot represent the company's opinions, the company does not agree with his openness, and you should not blame the author.
